Rename Frame to LocalFrame

cdumez opened the pull request:
#### 2461c9a4068d46d1786b27b98c280690c883f28d
<pre>
Rename Frame to LocalFrame
<a href="https://bugs.webkit.org/show_bug.cgi?id=253786">https://bugs.webkit.org/show_bug.cgi?id=253786</a>

Reviewed by Alex Christensen.

Rename Frame to LocalFrame, as a preparation for site isolation.

* Source/WebCore/*:

Canonical link: <a href="https://commits.webkit.org/261573@main">https://commits.webkit.org/261573@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ee1891f95c73cdfda7ecaaa4c6cfc8aeb0ec6770

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/112112 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/21244 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/90/builds/736 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/3909 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/120768 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/116173 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/22592 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/12364 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/86/builds/4217 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/117873 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/16782 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/99955 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/105170 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/98737 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/88/builds/500 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/45790 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/13655 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/89/builds/535 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/93414 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/14334 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/9904 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/19703 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/52534 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/8068 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/16125 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/accessibility/AccessibilityObject.cpp:
```diff
@@ -364,7 +364,7 @@ bool AccessibilityObject::hasMisspelling() const
         return false;
     
-    Frame* frame = node()->document().frame();
+    LocalFrame* frame = node()->document().frame();
```
How about `auto*` for all of the locals in this file?

darinadler commented on Source/WebCore/accessibility/AccessibilityObjectInterface.h:
```diff
@@ -78,7 +78,7 @@ class AXTextMarkerRange;
 class Document;
 class Element;
-class Frame;
+class LocalFrame;
 class FrameView;
```
Should re-sort these.

cdumez committed:
Rename Frame to LocalFrame
https://bugs.webkit.org/show_bug.cgi?id=253786

Reviewed by Alex Christensen.

Rename Frame to LocalFrame, as a preparation for site isolation.

* Source/WebCore/*:

Canonical link: https://commits.webkit.org/261573@main
